learn/experimental/vector_search.mdx

```diff
@@ -30,39 +30,139 @@ curl \
   }'
 ```
 
-### Generate vector embeddings using a third-party tool
+### Generate vector embeddings
 
-To use vector search, you must generate embeddings with a third-party tool such as [OpenAI](https://openai.com/), [HuggingFace](https://huggingface.co/), and [Cohere.AI](https://cohere.com/).
+To use vector search, first configure the `embedders` index setting. You may configure multiple embedders for an index. 
 
-<Capsule intent="note" title="What is an embedding?">
-Vector search relies on embeddings. Vector embeddings are a way of representing data that identify similarities in meaning and context.
+Embedders generate vector data from your documents. Meilisearch natively supports [OpenAI](https://openai.com/) and [HuggingFace](https://huggingface.co/) embedders. 
+
+It is also possible to use custom embedders. In this case, you must generate the embeddings manually and add include them as a field in your documents.
+
+#### Generate auto-embeddings with OpenAI 
+
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source":  "openAI",
+        "apiKey": "anOpenAiApiKey",
+        "model": "text-embedding-ada-002",
+        "documentTemplate": "A movie titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+      }
+    }
+  }'
+```
+
+`documentTemplate` is an optional field you can use to send only relevant data to the model. `documentTemplate` must be a [Liquid template](https://shopify.github.io/liquid/).
+
+#### Generate auto-embeddings with HuggingFace 
+
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source":  "huggingFace",
+        "model": "bge-base-en-v1.5",
+        "documentTemplate": "A movie titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+      }
+    }
+  }'
+```
+
+`documentTemplate` is an optional field you can use to send only relevant data to the model. `documentTemplate` must be a [Liquid template](https://shopify.github.io/liquid/).
```

Comment:
I think we need more information about the `documentTemplate` in a dedicated section, otherwise I don't think users will be able to use it.

Missing information:

1. The document template is like a text with blanks that are filled by the fields of each document during the indexing process, and serve as the view of the document that will be used for computing the embedding.
2. The document template should be written such that the template filled by documents produces a short text that is highly semantically relevant. In particular:
    1. Information applicable to any document of the index can be added: "a movie", "a book", "a product"
    2. Information from fields that are not semantic (id, release date) can be left out
    3. Fields that can get very long should be truncated after some number of [characters](https://shopify.github.io/liquid/filters/truncate/) or of [words](https://shopify.github.io/liquid/filters/truncatewords/)
3. Although a default template is provided, it is highly recommended to customize the template to suit your document, because the default template contains all the fields untruncated, and so will typically produce texts that are too long and not relevant enough.
4. Meilisearch exposes two bits of context to the template:
    1. For each field `attribute` in the document, `{{ doc.attribute }}` will resolve to the value of that field in the document. ⚠️ If an `attribute` field referenced in the template is undefined for a document, then an indexing error will be raised (it is OK to have the field be `null` though, it will result in a [`nil`](https://shopify.github.io/liquid/basics/types/#nil) value)
    2. The special `{{ fields }}` variable is an [array](https://shopify.github.io/liquid/basics/types/#array) that contain one `field` element per field appearing in a document from the dataset. Each `field` element has two subfields:
    - `{{ field.name }}`: name of that field (e.g. `text`)
    - `{{ field.value }}`: value of that field (e.g. `frying pan`), or [`nil`](https://shopify.github.io/liquid/basics/types/#nil) if the field is undefined or `null` for that document.

Reply:
This document is supposed to be more of a quick start with vector search than an exhaustive reference. Going into too much detail will make it lose focus as it tries to do too many things at once (advertise the feature, guide new users, serve as a reference for advanced users).

I will integrate some of what you say into this document because users won't have anywhere else to refer to (perhaps the GH discussion, if it's still being updated regularly), but I will probably only add everything to a separate document.

It's clear our current approach to experimental features is breaking down, which is something I plan to address very soon.

Reply:
> This document is supposed to be more of a quick start with vector search than an exhaustive reference. 

In general, I agree. Still I think using a custom template is crucial to correctly using the feature (I'm starting to regret having included a default template rather than force users to pass a template)